components/board/customlinux/setup.go

```diff
@@ -0,0 +1,83 @@
+// Package customlinux implements a board running linux
+package customlinux
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board/genericlinux"
+)
+
+// GenericLinuxPin describes a gpio pin on a linux board.
+type GenericLinuxPin struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"`
+	RelativeID      int    `json:"relative_id"`
+	PWMChipSysFSDir string `json:"pwm_chip_dir,omitempty"`
```

Comment:
Mild preference for the name in this struct and the name in the JSON to resemble each other. Maybe this should be `PwmChipDir`? Maybe the JSON key should be "pwm_chip_sys_fs_dir".